docs/enhancements/arm64-cpu-architecture.md

```diff
@@ -0,0 +1,152 @@
+            "rhcos_rootfs": "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest-4.9/rhcos-live-rootfs.x86_64.img",
+            "rhcos_version": "49.84.202107032011-0"
+      },
+      "arm64": {
+            "release_version": "4.9.0-0.nightly-arm64-2021-07-05-092650",
+            "release_image": "quay.io/openshift-release-dev/ocp-release-nightly@sha256:74493f2ad08a86f3f001ab398735965d0253755dbe4186ee7aa9e6d6a809ae3b",
+            "rhcos_image": "https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/pre-release/latest-4.9/rhcos-live.aarch64.iso",
+            "rhcos_rootfs": "https://mirror.openshift.com/pub/openshift-v4/aarch64/dependencies/rhcos/pre-release/latest-4.9/rhcos-live-rootfs.aarch64.img",
+            "rhcos_version": "49.84.202106272247-0"
+      }
+   }
+}
+```
+
+For backwards compatibility, the current format of openshift-version should still be supported.
+So older versions (<4.9), that don't support arm64 CPU architecture, could use the old format x86_64 architecture only.
+
+### Cluster creation and ISO generation
+
+When creating a new cluster, `cpu_architecture` property should be supported to define which hosts can be registered to it. This property would be added to cluster-create-params and stored in cluster upon creation. If not specified, defaulted to x86_64 as a fallback.
+
+An example for the swagger schema:
+
+```
+cpu_architecture:
+   type: string
+   default: 'x86_64'
+   description: The CPU architecture of the image (x86_64/arm64/etc)
+```
+
+When generating a discovery ISO, the proper RHCOS/Release images should be used according to CPU architecture specified in the cluster. The `cpu_architecture` property should be stored in `infraenv` after the ISO is generated.
+
+### CPU_ARCHITECTURES environment variable
+
+The supported CPU architectures will be defined in CPU_ARCHITECTURES environment variable. The default should be: 'x86_64,arm64'.
+As arm64 is not supported for older OCP versions (<4.9), it could be validated against OPENSHIFT_VERSIONS only for OCP >=4.9.
+
+### User Stories
+
+#### Story 1
+
+As an Assisted-Installer user, I need to install a cluster on non x86_64 hosts.
+
+#### Story 2
+
+As an Assisted-Installer user, I need to install a Single-Node OpenShift cluster on an arm64 host.
+
+### Implementation Details/Notes/Constraints [optional]
+
+* The current focus is for SNO flows, but other scenarios could be supported.
+
+* Late binding is not covered by this enhancement, but decoupling the discovery ISO generation from the cluster should be simple enough. I.e. adding `cpu_architecture` property to `infraenv_create_params` and override the architecture defined in the cluster. Also, architecture validation should be probably added to host discovery conditions (i.e. to ensure that the host CPU architecture is similar to the cluster's architecture).
+
+### Open Questions
+
```

Comment:
Do we need to do anything in order to accommodate CNV and LSO operators?
Users can enable them from AI UI, unsure what will be the result on ARM.

Reply:
@gamli75 do you know if these operators support arm? maybe we could just disable the installation on arm for now?

Reply:
Both are out of scope for the first OCP deliverable of arm support.

Reply:
Good call @eranco74 .  We have the following validations:
      - 'lso-requirements-satisfied'
      - 'ocs-requirements-satisfied'
      - 'cnv-requirements-satisfied'
We already check various cluster properties when deciding if those validations pass or fail.  We should add a check to ensure that the CPU architecture equals x86_64.